Sources/BothamViewController.swift

```diff
@@ -11,31 +11,34 @@ import Foundation
     import UIKit
 
 public class BothamViewController: UIViewController, BothamUI {
-    public var presenter: BothamPresenter! = nil
+    private var _presenter: BothamPresenter? = nil
 
     public override func viewDidLoad() {
         super.viewDidLoad()
-        presenter.viewDidLoad()
+        if self.respondsToSelector(Selector("presenter")) {
+            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
+        }
+        _presenter?.viewDidLoad()
```

Comment:
Do you have example, of more than one presenter in the same ViewController
make sense ?

On Mon, Apr 11, 2016 at 4:55 PM, Davide Mendolia <davide@karumi.com> wrote:

> We should be really carefull, because this will break 1.0 implementation
> and migration is not trivial.
>
>
> On Mon, Apr 11, 2016 at 4:52 PM, Pedro Vicente Gómez Sánchez <
> notifications@github.com> wrote:
>
>> In Sources/BothamViewController.swift
>> <https://github.com/Karumi/BothamUI/pull/40#discussion_r59218057>:
>>
>> >
>> >      public override func viewDidLoad() {
>> >          super.viewDidLoad()
>> > -        presenter.viewDidLoad()
>> > +        if self.respondsToSelector(Selector("presenter")) {
>> > +            _presenter = self.performSelector(Selector("presenter")).takeUnretainedValue() as? BothamPresenter
>> > +        }
>> > +        _presenter?.viewDidLoad()
>>
>> In the Android implementation we keep a list of presenters which
>> lifecycle is going to be linked and we throw an exception if the presenter
>> is null and the lifecycle has started.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Karumi/BothamUI/pull/40/files/b469cd2f2e6aeb0c19328dd1a57f079493a9a592#r59218057>
>>
>
>